pkg/collector/providers/kubelet.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubelet
+
+package providers
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+)
+
+const (
+	podAnnotationFormat = "service-discovery.datadoghq.com/%s."
+)
+
+// KubeletConfigProvider implements the ConfigProvider interface for the kubelet.
+type KubeletConfigProvider struct {
+	kubelet *kubelet.KubeUtil
+}
+
+// NewKubeletConfigProvider returns a new ConfigProvider connected to kubelet.
+// Connectivity is not checked at this stage to allow for retries, Collect will do it.
+func NewKubeletConfigProvider(config config.ConfigurationProviders) (ConfigProvider, error) {
+	return &KubeletConfigProvider{}, nil
+}
+
+func (k *KubeletConfigProvider) String() string {
```

Comment:
I'm wondering if this should be part of the interface, some provider have it some don't

Reply:
indeed, adding it to the interface and the ConsulProvider